src/libraries/System.Transactions.Local/tests/OleTxTests.cs

```diff
@@ -474,4 +477,45 @@ private static void Retry(Action action)
             }
         }
     }
+
+    public class OleTxFixture
+    {
+        public OleTxFixture()
+        {
+            if (RuntimeInformation.ProcessArchitecture == Architecture.X86)
+            {
+                return; // Temporarily skip on 32-bit where we have an issue
+            }
+
+            // In CI, we sometimes get XACT_E_TMNOTAVAILABLE on the very first attempt to connect to MSDTC;
+            // this is likely due to on-demand slow startup of MSDTC. Perform pre-test connecting with retry
+            // to ensure that MSDTC is properly up when the first test runs.
+            int nRetries = 5;
+
+            while (true)
```

Comment:
Do we need to wait at all between retries, or trying immediately is sufficient?

Reply:
Trying immediately seems to be sufficient - when we see a CI failure it's always that one very first test that fails, the second works. But I'll add a 100ms delay, just in case...